File: app/toes/processing_node.py
from typing import Dict

from app.toes.node import Node


class ProcessingNode(Node):
    def __init__(self, parent: Node, name: str = "", attributes: Dict = {}):
        super(ProcessingNode, self).__init__(
            name=name,
            parent=parent,
            paired_tag=False,
            attributes=attributes
        )
        self.type = Node.PROCESSING

File: src/main/scala/com/sloth/templating/nodes/ProcessingNode.scala
package com.sloth.templating.nodes

import scala.collection.mutable
import scala.collection.mutable.ListBuffer

class ProcessingNode(name: String = "", children: ListBuffer[Node] = ListBuffer(), parent: Node, attributes: mutable.HashMap[String, String] = mutable.HashMap())
  extends Node(name = name, children = children, argPairedTag = false, nodeType = Node.PROCESSING, parent = parent, attributes = attributes) {

}
